PEPRest/src/main/java/it/cnr/iit/peprest/PEPRest.java

```diff
@@ -179,8 +187,8 @@ public String receiveResponse( Message message ) {
         Reject.ifNull( message );
         try {
             responsesMap.put( message.getMessageId(), message );
-            messageStorage.addMessage( message );
             unansweredMap.remove( message.getMessageId() );
+            messageStorage.addMessage( message );
```

Comment:
what effect resolve this reordering of addition?

Reply:
none. 
before we operate on maps and then on the messageStore